samples/snippets/authorized_view_tutorial.py

```diff
@@ -27,16 +27,18 @@ def run_authorized_view_tutorial(override_values={}):
 
     client = bigquery.Client()
     source_dataset_id = "github_source_data"
+    source_dataset_id_full = "{}.{}".format(client.project, source_dataset_id)
 
     # [END bigquery_authorized_view_tutorial]
     # [END bigquery_avt_create_source_dataset]
     # To facilitate testing, we replace values with alternatives
     # provided by the testing harness.
     source_dataset_id = override_values.get("source_dataset_id", source_dataset_id)
+    source_dataset_id_full = "{}.{}".format(client.project, source_dataset_id)
```

Comment:
Not super familiar with the API, but should there be a helper for this?

Reply:
`client.dataset` was that helper, but we had so many people confused why it didn't get the Dataset metadata (i.e. `get_dataset`), that we deprecated it.

A dot-separate string maps more closely to what BQ developers expect from SQL and the BQ GUI.

I've had thoughts of allowing a `Dataset` object to have `None` as a project ID, but that'd be pretty error-prone, given all the current logic expects a project ID on those objects.

https://github.com/googleapis/google-cloud-python/issues/8989